packages/eas-cli/src/credentials/ios/actions/AscApiKeyUtils.ts

```diff
@@ -1,15 +1,123 @@
 import chalk from 'chalk';
 import fs from 'fs-extra';
+import { nanoid } from 'nanoid';
 import path from 'path';
-import * as uuid from 'uuid';
 
 import Log, { learnMore } from '../../../log';
-import { promptAsync } from '../../../prompts';
-import { getCredentialsFromUserAsync } from '../../utils/promptForCredentials';
+import { confirmAsync, promptAsync } from '../../../prompts';
+import { CredentialsContext } from '../../context';
+import {
+  getCredentialsFromUserAsync,
+  shouldAutoGenerateCredentialsAsync,
+} from '../../utils/promptForCredentials';
 import { AscApiKey } from '../appstore/Credentials.types';
-import { AscApiKeyPath, MinimalAscApiKey, ascApiKeyMetadataSchema } from '../credentials';
+import {
+  AscApiKeyPath,
+  MinimalAscApiKey,
+  ascApiKeyIdSchema,
+  ascApiKeyIssuerIdSchema,
+} from '../credentials';
+import { isAscApiKeyValidAndTrackedAsync } from '../validators/validateAscApiKey';
 
-export async function promptForAscApiKeyAsync(): Promise<AscApiKeyPath> {
+export enum AppStoreApiKeyPurpose {
```

Comment:
Do we need this enum at all? When do we use the `UNKOWN` value?

Reply:
I got rid of the unknown value - i ended up not using it anywhere 😅 . I'd rather keep the `AppStoreApiKeyPurpose` enum rather than hard coding asc api keys for submissions because: 
- asc api keys can be used for a lot of things, not just submissions. There is a good chance we may use it for other purposes in the future. We ended up doing smth similar in android with service account keys.
- Apple Api will reject names that are >30 chars, i wrote a regression test loops through the enums to make sure that whatever we add in the future wont break
- there's a lot of other places in the code that will break if we hardcode for submissions right now, then decide to add another use case. Currently, either typescript will catch it cuz it's in a switch statement, or it will fail with a 'this purpose is not supported yet' error, instead of a cryptic error.